### ISSUE TITLE
Research implementing Apple Musics Web Player

### ISSUE THREAD

**Issue by toastmalone** [wethegreenpeople/PartyPlaylists#35]
https://developer.apple.com/documentation/musickitjs

For now, just focus on getting it working in the PartyPlaylists.MVC

**Comment by wethegreenpeople** [comment]
Considerations:
- Tables need to be updated with AM auth codes
- Authorizations per music service
- Searching between multiple services (spotify/apple music), multiple search boxes, one search box, multiple distinguishing icons?
- Swapping between active web players (when playing a spotify song use spotify player, when playing AM song use Apple player)

**Comment by toastmalone** [comment]
Yes, I believe searching between all authorized services might be useful. We will have to keep in mind getting back multiples of the same results from multiple services. Filtering by uniqueness and response time could be a nice way to handle that. Ideally, one search box is what I was shooting for.

**Comment by RickyGenz** [comment]
I would like to fill in `AppleMusicService.cs`. Would that be useful to you guys?

**Comment by wethegreenpeople** [comment]
Yeah that would be awesome! You can clone the master branch and start working on it, with (hopefully) just a couple of setup steps:

You'll need to implement an appsettings.json file in the PartyPlaylists and PartyPlaylists.MVC project. You can use the same file for both projects, but you'll have to have it. The file looks like so:
```{
  "Logging": {
    "LogLevel": {
      "Default": "Debug",
      "System": "Information",
      "Microsoft": "Information"
    }
  },
  "ConnectionStrings": {
    "PartyPlaylistConnectionString": "<CONNECTION_STRING>"
  },
  "Jwt": {
    "Key": "<LITERALLY_ANY_STRING_BUT_WE_TYPICALLY_USE_GUIDS>"
  },
  "SpotifyAuthRedirectUri": "https://localhost:44388/Room/SpotifyAuthorized/"
}
```
You can hit me up on Discord, or join our server https://discord.gg/cdmx8VV and I can give you the live DB connection string. Theoretically you should also be able to do everything within a local DB if you wanted/needed.

You shouldn't need to, but if you need Spotify app authentications let me know and we can get you a set of those as well.
